core/trino-main/src/main/java/io/trino/execution/buffer/OutputBufferMemoryManager.java

```diff
@@ -110,6 +121,9 @@ public void updateMemoryUsage(long bytesAdded)
                     this.bufferBlockedFuture = null;
                 }
             }
+            long recordTime = ticker.read();
+            outputBufferUtilization.add(getUtilization(), (double) (recordTime - this.lastBufferUtilizationRecordTime));
+            this.lastBufferUtilizationRecordTime = recordTime;
         }
```

Comment:
You should also update utilization in `io.trino.execution.buffer.OutputBufferMemoryManager#close`. Could you check that such measurement is actually captured in final operator stats?

Reply:
> Could you check that such measurement is actually captured in final operator stats? 
What do you mean? 
If you mean wheter this one is treated as a final stats - yes it is. It is not stripped on the task completion.
If you man "how often" a supplier is asked to provide this stats - it is very often on different events (like creating / deleting tasks and so on).

Reply:
I mean if last `recordOutputBufferUtilization()` that happens in `close` is actually accounted for in final stats

Reply:
No, it is not accounted. It looks like lifetime of `outputBuffer` is decoupled from lifetime of `operator` and that is a reason that we are not able observe that event in that approach (without making a kind of skeweness).

Reply:
ping https://github.com/trinodb/trino/pull/13463#discussion_r952373652